akka-http-core/src/main/scala/akka/http/engine/client/ConnectionPool.scala

```diff
@@ -0,0 +1,141 @@
+      RequestContext   |           +----+      |                   |    +----> |               |  ResponseContext
+    +----------------> | Conductor |---------> | Connection Slot 2 +---------> | responseMerge +------------------>
+                       |           +----+      |                   |    +----> |               |
+                       +-----------+    |      +---------+---------+    |      +---------------+
+                             ^          |          |     |              |                        
+                             |          |      +-------------------+    |                        
+                             |          |      |                   |    |                        
+                   SlotEvent |          +----> | Connection Slot 3 +---->                        
+                             |                 |                   |                             
+                             |                 +---------------+---+                             
+                             |                     |     |     |                                    
+                       +-----------+    SlotEvent  |     |     | 
+                       | slotEvent | <-------------+     |     |
+                       |   Merge   | <-------------------+     |                                  
+                       |           | <-------------------------+                                  
+                       +-----------+
+
+    Conductor:
+    - Maintains slot state overview by running a simple state machine per Connection Slot
+    - Decides which slot will receive the next request from upstream according to current slot state and dispatch configuration
+    - Forwards demand from selected slot to upstream
+    - Always maintains demand for SlotEvents from the Connection Slots
+    - Implemented as a sub-graph
+
+    Connection Slot:
+    - Wraps a low-level outgoing connection flow and (re-)materializes and uses it whenever necessary
+    - Directly forwards demand from the underlying connection to the Conductor
+    - Dispatches SlotEvents to the Conductor (via the SlotEventMerge)
+    - Implemented as a sub-graph
+
+    Response Merge:
+    - Simple merge of the Connection Slots' outputs
+
+  */
+  private def newPoolFlow(connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]],
+                          remoteAddress: InetSocketAddress, settings: HostConnectionPoolSettings, log: LoggingAdapter)(
+                            implicit system: ActorSystem, fm: FlowMaterializer): Flow[RequestContext, ResponseContext, Any] =
+    Flow() { implicit b ⇒
+      import settings._
+      import FlowGraph.Implicits._
+
+      val conductor = b.add(HostConnectionPoolConductor(maxConnections, maxRetries, pipeliningLimit, log))
+      val slots = Vector
+        .tabulate(maxConnections)(HostConnectionPoolSlot(_, connectionFlow, remoteAddress, settings))
+        .map(b.add(_))
```

Comment:
I believe not.
But it's certainly good to take a moment in quiet content,
reflect on the beauty of what was given to us,
and praise the creator.

And then jump right back into the flame war surrounding `CanBuildFrom`.